components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.casper.CasperConstants;
+import org.apache.camel.component.jackson.JacksonDataFormat;
+import org.apache.camel.impl.DefaultCamelContext;
+
+import com.syntifi.casper.sdk.model.auction.AuctionState;
+import com.syntifi.casper.sdk.model.balance.BalanceData;
+import com.syntifi.casper.sdk.model.block.JsonBlock;
+import com.syntifi.casper.sdk.model.era.JsonEraValidators;
+import com.syntifi.casper.sdk.model.stateroothash.StateRootHashData;
+
+/**
+ * Demo Class loading Casper components example routes  
+ * @author p35862
+ *
+ */
+
+@SuppressWarnings("deprecation")
+public class DemoApp {
+
+	/**
+	 * loads route 1
+	 * @param cntxt : Camel context
+	 * @param temp : producerTemplate
+	 * @throws Exception : exception
+	 */
+	private static void loadroute1(CamelContext cntxt, ProducerTemplate temp) throws Exception {
+		cntxt.addRoutes(new RouteBuilder() {
+			public void configure() throws Exception {
+				from("direct:" + CasperConstants.STATE_ROOT_HASH).routeId("STATE_ROOT_HASH")
+						.to("casper:http://65.21.227.180:7777/?operation=" + CasperConstants.STATE_ROOT_HASH)
+						.process(new Processor() {
+
+							@Override
+							public void process(Exchange exchange) throws Exception {
+								StateRootHashData state = (StateRootHashData) exchange.getIn().getBody();
+								System.err.println("* Current STATE_ROOT_HASH is : " + state.getStateRootHash());
+							}
+						});
+			}
+		});
+
+		cntxt.start();
+		temp.sendBody("direct:" + CasperConstants.STATE_ROOT_HASH, "This is a test message");
+		cntxt.stop();
+
+	}
+
+	/**
+	 * loads route 2
+	 * @param cntxt : Camel context
+	 * @param temp : producerTemplate
+	 * @throws Exception : exception
+	 */
+	private static void loadroute2(CamelContext cntxt, ProducerTemplate temp) throws Exception {
+		cntxt.addRoutes(new RouteBuilder() {
+			public void configure() throws Exception {
+				from("file:src/main/resources/datas/?fileName=get_block.txt&charset=utf-8&noop=true")
+						.convertBodyTo(String.class).routeId(CasperConstants.BLOCK).setHeader("BLOCK_HASH", body())
+						.to("casper:http://65.21.227.180:7777/?operation=" + CasperConstants.BLOCK)
+
+						.process(new Processor() {
+
+							@Override
+							public void process(Exchange exchange) throws Exception {
+								JsonBlock block = (JsonBlock) exchange.getIn().getBody();
+								String blockHash = (String) exchange.getIn().getHeader("BLOCK_HASH");
+								System.err.println("* getBlock was called with parameter block Hash = " + blockHash);
+								System.err.println("* getBlock retrieved a block that was minted at era ="
+										+ block.getHeader().getEraId() + " and has as parent hash = "
+										+ block.getHeader().getParentHash());
+
+							}
+						});
+
+			}
+
+		});
+
+		cntxt.start();
+		Thread.sleep(5000);
```

Comment:
Please avoid using `Thread.sleep` as much as possible. Instead, please try to replace the code with Awaitility or, in case of synchronization / concurrency, latches and other concurrency handling mechanisms.